### ISSUE TITLE
Auto-Escape String Delimiters in String Literals

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#5009]
**Justification**
SCP is currently bailing out when it determines that the caret is inside a string literal. There's an opportunity to handle double-quote escaping here - something that is still very frequently being asked about on SO.

**Description**
Instead of bailing out, handle `"` keypress inside a string literal by doubling the character and placing the caret *after* the escaped quotes:

    MsgBox "foo|bar" ~> MsgBox "foo""|bar"

One consideration is the closing character - SCP behavior needs to remain consistent, so if the caret is just before the string-closing quote, auto-escape should not happen (i.e. preserve current behavior):

    MsgBox "foobar|" ~> MsgBox "foobar"|

Adding escaped double-quotes at the end of a string literal would need to be done by adding a temporary space, that can then be removed:

    MsgBox "foobar| " ~> MsgBox "foobar"" "

**Comment by Vogel612** [comment]
Suggestion: instead of the temporary space workaround, why not have the following behaviour?

```
MsgBox "foobar|" ~> MsgBox "foobar"| ~> MsgBox "foobar""|"
```

This seems a bit more intuitive.

**Comment by retailcoder** [comment]
@Vogel612 first to second is the current behavior; first to third would be the new one? I don't think second to third is intuitive at all (caret moving *back*?)... or I'm misreading the example?

**Comment by retailcoder** [comment]
Yeah I misread it: I think we'd get `MsgBox "foobar"| ~> MsgBox "foobar""|"` for free, without any changes to the current logic.
